akka-docs/rst/java/code/docs/http/javadsl/server/directives/CustomHttpMethodExamplesTest.java

```
@@ -0,0 +1,86 @@
+import java.util.concurrent.TimeUnit;
+
+import static akka.http.javadsl.model.HttpProtocols.HTTP_1_0;
+
+public class CustomHttpMethodExamplesTest extends JUnitRouteTest {
+
+  @Test
+  public void testComposition() {
+    ActorSystem  system = system();
+    Materializer materializer = materializer();
+    LoggingAdapter loggingAdapter = NoLogging$.MODULE$;
+
+    int    port = 9090;
+    String host = "127.0.0.1";
+
+    //#customHttpMethod
+    HttpMethod BOLT =
+      HttpMethods.custom("BOLT", false, true, RequestEntityAcceptance.Expected$.MODULE$);
+    final ParserSettings parserSettings =
+      ParserSettings.create(system).withCustomMethods(BOLT);
+    final ServerSettings serverSettings =
+      ServerSettings.create(system).withParserSettings(parserSettings);
+
+    final Route routes = route(
+      extractMethod( method ->
+        complete( "This is a " + method.name() + " request.")
+      )
+    );
+    final Flow<HttpRequest, HttpResponse, NotUsed> handler = routes.flow(system, materializer);
+    final Http http = Http.get(system);
+    final CompletionStage<ServerBinding> binding =
+      http.bindAndHandle(
+        handler,
+        ConnectHttp.toHost(host, port),
+        serverSettings,
+        loggingAdapter,
+        materializer);
+
+    HttpRequest request = HttpRequest.create()
+      .withUri("http://" + host + ":" + Integer.toString(port))
+      .withMethod(BOLT)
+      .withProtocol(HTTP_1_0);
+
+    CompletionStage<HttpResponse> response = http.singleRequest(request, materializer);
+
+    try {
+      assertResult(StatusCodes.OK, response.toCompletableFuture().get().status());
+    } catch (Exception e) {
+      fail(e);
+    }
+
+    try {
+      assertResult(
+        "This is a BOLT request.",
+        response.toCompletableFuture().get().entity().toStrict(3000, materializer).toCompletableFuture().get().getData().utf8String()
+      );
+    } catch (Exception e) {
+      fail(e);
+    }
```

why this construction? Just let it throw